tools/profiles/debug.json

```diff
@@ -22,6 +22,16 @@
         "cxx": ["--cpp", "--no_rtti", "--no_vla"],
         "ld": []
     },
+    "uARM": {
```

Comment:
@bridadan @theotherjimmy 
Why did we add another "toolchain" in order to use MicroLib instead of having something [like this](https://github.com/ARMmbed/mbed-os/blob/master/tools/toolchains/gcc.py#L57) in [arm.py](https://github.com/ARMmbed/mbed-os/blob/master/tools/toolchains/arm.py)?

Reply:
@hugueskamba The `uARM` toolchain was present in Mbed OS 2.0, so it would have been a breaking change to remove it. In Mbed OS 5.0 the `default_lib` parameter was added to GCC (and maybe IAR as well?). I always hoped we would eventually deprecate the `uARM` toolchain and use the `default_lib` in its place!

Reply:
Thanks @bridadan ,
Wouldn't removing `uARM` cause other changes? The first thing I see is the fact that the [boot code](https://github.com/ARMmbed/mbed-os/tree/master/rtos/source/TARGET_CORTEX) is different. However, would it just be as simple as moving the stuff from `mbed_boot_arm_micro.c` to `mbed_boot_arm_std.c`, renaming `mbed_boot_arm_std.c` to `mbed_boot_arm.c` and adding a bunch of macro?

Reply:
Yup like you said its more complicated than just changing the tools. I think that's one of the main reasons we didn't do it at the time. Changing all the device code would be a big effort that's for sure.

But yeah I think that's the general gist. I have to admit I'm not as familiar with the boot sequence requirements for `uARM` so I may be missing a few details.